pkg/server/cloud/cloud.go

```diff
@@ -70,6 +80,14 @@ type CloudClient interface {
 	IsAvailable() (bool, error)
 	AddInstanceParameter(instanceID, key, value string, isSecret bool) error
 	DeleteInstanceParameter(instanceID, key string) error
+
+	// Because a boot image spec can match multiple architectures we turn it into
+	// a list. Refactor for later: we should probably use different types.
+	Extend(spec BootImageSpec) []BootImageSpec
```

Comment:
If you look at naming in this interface, all the function names are descriptive of what they are supposed to do. `Extend` is very ambiguous, I think it'd be better to call this `SplitBootImageSpec` or something similar.